python_on_whales/components/pod/cli_wrapper.py

```diff
@@ -0,0 +1,778 @@
+
+    def exists(self, pod: ValidPod) -> bool:
+        """Returns `True` if the pod exists. `False` otherwise.
+
+         It's just calling `docker.pod.inspect(...)` and verifies that it doesn't throw
+         a `python_on_whales.exceptions.NoSuchPod`.
+
+        # Returns
+            A `bool`
+        """
+        try:
+            self.inspect(pod)
+        except NoSuchPod:
+            return False
+        else:
+            return True
+
+    @overload
+    def inspect(self, x: ValidPod, /) -> Pod:
+        ...
+
+    @overload
+    def inspect(self, x: List[ValidPod], /) -> List[Pod]:
+        ...
+
+    def inspect(
+        self, x: Union[ValidPod, Sequence[ValidPod]], /
+    ) -> Union[Pod, List[Pod]]:
+        """Creates a `python_on_whales.Pod` object.
+
+        # Returns
+            `python_on_whales.Pod`, or `List[python_on_whales.Pod]` if the input
+            was a list of strings.
+
+        # Raises
+            `python_on_whales.exceptions.NoSuchPod` if one of the pods does not exist.
+
+        """
+        if isinstance(x, list):
+            return [Pod(self.client_config, identifier) for identifier in x]
+        else:
+            return Pod(self.client_config, x)
+
+    def kill(
+        self,
+        x: Optional[Union[ValidPod, Sequence[ValidPod]]] = None,
+        /,
+        *,
+        all: bool = False,
+        latest: bool = False,
+        signal: Optional[Union[int, str]] = None,
+    ) -> None:
+        """Kill pods.
+
+        Parameters:
+            x: One or more pods to kill
+            all: Kill all pods
+            latest: Kill the latest pod
+            signal: The signal to send the pods' containers
+
+        # Raises
+            `python_on_whales.exceptions.NoSuchPod` if a pod does not exist.
+
+        """
+        pods = to_list(x)
+        if len(pods) == 0 and not latest and not all:
+            return
+
+        full_cmd = self.docker_cmd + ["pod", "kill"]
+
+        full_cmd.add_flag("--all", all)
+        full_cmd.add_flag("--latest", latest)
+        full_cmd.add_simple_arg("--signal", format_signal_arg(signal))
+        full_cmd.extend([str(p) for p in pods])
+
+        run(full_cmd)
+
+    def list(self, *, filters: PodListFilters = {}) -> List[Pod]:
+        """List the pods on the host.
+
+        Parameters:
+            filters: Filters to apply when listing pods
+
+        # Returns
+            A `List[python_on_whales.Pod]`
+        """
+        full_cmd = self.docker_cmd + ["pod", "ps", "-q", "--no-trunc"]
+        full_cmd.add_args_list("--filter", format_dict_for_cli(filters))
+
+        return [
+            Pod(self.client_config, x, is_immutable_id=True)
+            for x in run(full_cmd).splitlines()
+        ]
+
+    def logs(
+        self,
+        pod: ValidPod,
+        *,
+        container: Optional[
+            python_on_whales.components.container.cli_wrapper.ValidContainer
+        ] = None,
+        names: bool = False,
```

Comment:
```suggestion
        /,
        container: Optional[
            python_on_whales.components.container.cli_wrapper.ValidContainer
        ] = None,
        *,
        names: bool = False,
```
To be consistent with the other comment.

Reply:
I don't think we should make the `pod` argument positional-only, there's nothing wrong with a user doing `podman.pod.logs(container="my-ctr", pod="my-pod")`, unlike other methods which accept one or more pods so neither "pod" nor "pods" are always correct.